test/test_nestedtensor.py

```diff
@@ -7818,131 +7822,539 @@ def __torch_dispatch__(self, func, types, args=..., kwargs=None):
         self.assertEqual(res.shape, (4, nt.shape[1], 6))
 
 
-FORWARD_FAILURES = {
-    # === BEGIN NotImplementedError SECTION ===
-    # unary
-    "nn.functional.celu",
-    "nn.functional.elu",
-    "nn.functional.hardshrink",
-    "nn.functional.hardsigmoid",
-    "nn.functional.hardtanh",
-    "nn.functional.logsigmoid",
-    "nn.functional.mish",
-    "nn.functional.relu6",
-    "nn.functional.rrelu",
-    "nn.functional.selu",
-    "nn.functional.softplus",
-    "nn.functional.softshrink",
-    "nn.functional.threshold",
-    "rad2deg",
-    # binary
-    "__rsub__",
-    "complex",
-    "floor_divide",
-    "polar",
-    "rsub",
-    # reduction
-    "count_nonzero",
-    "linalg.vector_norm",
-    "nansum",
-    "std",
-    "std.unbiased",
-    "var",
-    "var.unbiased",
-    # === BEGIN UNSUPPORTED SECTION ===
-    # ValueError: expects strided tensor (got torch.jagged tensor)
-    "masked.amax",
-    "masked.amin",
-    "masked.argmax",
-    "masked.argmin",
-    "masked.logsumexp",
-    "masked.mean",
-    "masked.norm",
-    "masked.prod",
-    "masked.std",
-    "masked.sum",
-    "masked.var",
-    # === BEGIN BUG SECTION ===
-    # Returns a tuple of Tensors so it doesn't work with NJT's unary pointwise logic
-    "frexp",
+FORWARD_FAILURES = [
```

Comment:
Maybe worth adding a short comment on how you tend to think about organizing the xfails, or like a table of contents? :P

Reply:
Reading some more, table of contents seems unrealistic lol

Reply:
> Maybe worth adding a short comment on how you tend to think about organizing the xfails

I think this is a good point and something I should do here. You've managed to uncover a sorta tricky detail here - rules are attempted to be matched in list order, so higher-up xfails have higher precedent, so order actually matters! I'm not sure this is actually that great of a property, as it can be hard to maintain these in the right order. So some possibilities here:
1. Keep as-is and describe the ordering property, etc.
2. Always try to match all rules and fail if multiple rules match to ensure rules are mutually exclusive (this can make them slightly harder to write).
3. ???

Reply:
Ah that is tricky..
(2) is actually a great property, being able to see the error message and being able to reliably map it to a single bug means our error messages are good, but I guess the issue is here is that you might run into some generic compiler error message that you don't have control over and is always going to be ambiguous.

Reply:
Hmm one way to do it incrementally could be to have a `mutually_exclusive` property that is by default True, but False for the ones that are ambiguous.

Reply:
> (2) is actually a great property, being able to see the error message and being able to reliably map it to a single bug means our error messages are good and descriptive, but I guess the issue is here is that you might run into some generic compiler error message that you don't have control over and is always going to be ambiguous.

I think to clarify some: we should always be able to write rules that are not ambiguous wrt how they match. That is, matching can be completely mutually exclusive so no single sample matches to multiple rules, but this can make the rules require detailed checks to ensure this property.

Whether or not error messages are clear is kinda orthogonal to rule matching, although also a good point that it's a nice property to be able to search the XFail db for an encountered error message.